File: mobile/src/services/system.ts
import { apiFetch } from "../lib/api";

export interface SystemSettings {
    maintenance_mode: boolean;
    debug_mode: boolean;
    ota_updates_enabled: boolean;
    push_notifications_enabled: boolean;
    cache_version: string;
    last_sync: string;
}

export const systemService = {
    getSettings: async (): Promise<SystemSettings> => {
        return apiFetch<SystemSettings>("core", "/admin/system/settings");
    },

    updateSettings: async (data: Partial<SystemSettings>): Promise<SystemSettings> => {
        return apiFetch<SystemSettings>("core", "/admin/system/settings", {
            method: "PATCH",
            body: JSON.stringify(data),
        });
    },

    syncDatabase: async (): Promise<{ message: string }> => {
        return apiFetch<{ message: string }>("core", "/admin/system/database/sync", {
            method: "POST",
        });
    },

    clearCache: async (): Promise<{ message: string }> => {
        return apiFetch<{ message: string }>("core", "/admin/system/cache/clear", {
            method: "POST",
        });
    },
};

File: backend/michicondrias_core/app/api/routes/system.py
from typing import Any, Dict
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.api import deps
from app.db.session import get_db
from app.models.user import User

router = APIRouter()

_system_settings = {
    "maintenance_mode": False,
    "debug_mode": False,
    "ota_updates_enabled": True,
    "push_notifications_enabled": True,
    "cache_version": "1.0.0",
    "last_sync": ""
}


@router.get("/admin/system/settings")
def get_system_settings(
    current_user: User = Depends(deps.require_role("admin")),
) -> Any:
    """Get system settings (Admin only)."""
    return _system_settings


@router.patch("/admin/system/settings")
def update_system_settings(
    settings_update: Dict[str, Any],
    current_user: User = Depends(deps.require_role("admin")),
) -> Any:
    """Update system settings (Admin only)."""
    _system_settings.update(settings_update)
    return _system_settings


@router.post("/admin/system/database/sync")
def sync_database(
    current_user: User = Depends(deps.require_role("admin")),
) -> Any:
    """Trigger database sync (Admin only)."""
    from datetime import datetime
    _system_settings["last_sync"] = datetime.utcnow().isoformat()
    return {"message": "Database sync initiated successfully"}


@router.post("/admin/system/cache/clear")
def clear_cache(
    current_user: User = Depends(deps.require_role("admin")),
) -> Any:
    """Clear system cache (Admin only)."""
    from datetime import datetime
    _system_settings["cache_version"] = str(int(_system_settings.get("cache_version", "0").split(".")[0]) + 1) + ".0.0"
    _system_settings["last_sync"] = datetime.utcnow().isoformat()
    return {"message": "Cache cleared successfully"}
